dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/thread/BackPressureExecutorService.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.thread
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.configuration.BackPressureStrategy
+import com.datadog.android.core.thread.FlushableExecutorService
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+internal class BackPressureExecutorService(
+    val logger: InternalLogger,
+    backpressureStrategy: BackPressureStrategy
+) : ThreadPoolExecutor(
+    CORE_POOL_SIZE,
+    CORE_POOL_SIZE,
```

Comment:
we need to indicate somehow that it is a single thread executor service - it is possible to understand that right now only by looking on the source code of this class.